crates/rpc-types-engine/src/attributes.rs

```diff
@@ -26,6 +26,11 @@ pub struct OptimismPayloadAttributes {
         serde(skip_serializing_if = "Option::is_none", with = "alloy_serde::quantity::opt")
     )]
     pub gas_limit: Option<u64>,
+    /// If set, this sets the EIP-1559 parameters for the block.
+    ///
+    /// Prior to Holocene activation, this field should always be [None].
+    #[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]
+    pub eip_1559_params: Option<B64>,
```

Comment:
could we add a serde roundtrip test where some fields are set to none?

suddenly no longer 100% sure whether we don't actually need serde(default) here

Reply:
+1

Reply:
Though I don't think we do need a `serde(default)` since this follows the above optional fields.